### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Vertiq-Ben**
- Updates for 40-06 speed, 81-08. 23-06. and pulsing

### CODE REVIEW

**Review by iq-fred — APPROVED** [review]

File: docs/clients/brushless_drive.rst [hunk 1]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 1      | drive_phase_pwm               | set            | float     | pwm      | Open loop (gimbal) mode with this trottle [-1, 1]. Use with phase_angle.                                                                                                                                                                                                                                                                                                                                    |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:113]: trottle typo

File: docs/clients/brushless_drive.rst [hunk 2]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 19     | commutation_hz                | get, set, save | uint32    | Hz       | Frequency of commutation. Higher frequencies run faster and more efficient, but may not give the controller enough computation time.                                                                                                                                                                                                                                                                        |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:149]: efficiently

File: docs/clients/brushless_drive.rst [hunk 3]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 46     | slew_volts_per_second         | get, set, save | float     | V/s      | The maximum allowed rate of change for the drive voltage when the slew rate is enabled. E.g. If this 100, Then the drive voltage of the motor cannot change faster than 100 V/s.                                                                                                                                                                                                                            |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:195]: If this is 100

**Review by vertiq-jordan — CHANGES_REQUESTED** [review]
Some typos in there we should fix before this goes into master

File: docs/clients/iquart_flight_controller_interface.rst [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+A minimal working example for the IQUART Flight Controller Interface Client is:
+
+.. code-block:: Python
+    :substitutions:
+    
+    import iqmotion as iq
+
+    com = iq.SerialCommunicator("/dev/ttyUSB0")
+    |variable_name| = iq.|module_name|(com, 0)
```
Comment (iq-matt) [inline docs/clients/iquart_flight_controller_interface.rst:112]: |variable_name| and |module_name| are showing up as those literal strings.  They should be filled in.

**Comment by iq-fred** [comment]
@vertiq-jordan and @iq-matt Take another look at this now that Ben has updated it

File: docs/clients/brushless_drive.rst [hunk 5]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 8      | drive_pwm                     | get            | float     | pwm      | The applied pwm after all computation and limiting [-1, 1]                                                                                                                                                                                                                                                                                                                                                  |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:127]: I think pwm should generally be PWM wherever it shows up in here when it's not a parameter name

File: docs/clients/brushless_drive.rst [hunk 6]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 9      | drive_volts                   | get            | float     | V        | The applied pwm after all computation and limiting                                                                                                                                                                                                                                                                                                                                                          |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:129]: pwm -> PWM

File: docs/clients/brushless_drive.rst [hunk 7]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 14     | motor_pole_pairs              | get, set, save | uint16    | PP       | Number of motor pole pairs (magnets/2)                                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:139]: do we need to specify the (magnets/2) part? if we do, can we change it to (number of rotor magnets / 2)? @iq-matt

File: docs/clients/brushless_drive.rst [hunk 8]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 48     | motoring_supply_current_limit | get, set, save | float     | A        | On modules that support dynamic supply current limiting, this entry determines the current limit for the current from the supply when the module is "motoring", i.e. the drive voltage is greater than the back-EMF, as opposed to when it is regenerating. Note this is the current from the supply, so it would be the current from your battery or power supply.                                         |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:199]: "motoring,"

File: docs/clients/brushless_drive.rst [hunk 9]
```diff
@@ -105,121 +105,121 @@ Message Table
...
++--------+-------------------------------+----------------+-----------+----------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 53     | motoring_limiting             | get            | uint8     | bool     | When dynamic supply current limiting is enabled, this reports whether the drive voltage is currently being limited due to the motoring supply current. True means that the limiter is currently controlling the drive voltage because the supply current is at or above the limit, false means the supply current is low enough that the limiter has not activated.                                         |
```
Comment (vertiq-jordan) [inline docs/clients/brushless_drive.rst:209]: can we come up with a better descriptor than "motoring." We use it a bunch of places, but it doesn't really mean anything outside of _Sister Christian_ (which is why we put it in quotes the first time).

File: docs/clients/coil_temperature_estimator_full.rst [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+| 7      | h_coil_amb_forced_conv_coeff | get, set, save | float     |              | Forced convection coefficient calculation coefficient, h_forced_conv = h_conv_coeff * sqrt(speed)                       |
++--------+------------------------------+----------------+-----------+--------------+-------------------------------------------------------------------------------------------------------------------------+
+| 8      | otw                          | get, set, save | float     | degC         | Over temperature warning, Derating of the motor begins at this temperature                                              |
++--------+------------------------------+----------------+-----------+--------------+-------------------------------------------------------------------------------------------------------------------------+
+| 9      | otlo                         | get, set, save | float     | degC         | Over temperature lock out, Derating of the motor end at this temperature, where the motor is fully disabled             |
```
Comment (vertiq-jordan) [inline docs/clients/coil_temperature_estimator_full.rst:137]: ends

File: docs/clients/brushless_drive.rst (hunk 9 above)
Comment (iq-fred) [reply docs/clients/brushless_drive.rst:209]: At this point, I think we are too far gone on the use of motoring. It does mean something to us, it means you are acting as a motor and not a generator. We've baked into the names of the parameters even. That whole terminology change is beyond the scope of this task.

File: docs/clients/coil_temperature_estimator_full.rst [hunk 11]
```diff
@@ -0,0 +1,158 @@
+Coil Temperature Estimator
+--------------------------
+
+The Coil Temperature Estimator is a convection and conduction based thermal model to estimate the temperature
+of motor coils when they are not directly sensed.  The temperature is used to derate the motor if the 
+temperature rises into dangerous levels. The temperature limits can be adjusted, though this is not recommended.
+The convection coefficient should be tuned for the specific propeller used on the motor.  A convection 
+coefficient for an average sized propeller for the given motor is loaded by default.  Consider changing the
```
Comment (vertiq-jordan) [inline docs/clients/coil_temperature_estimator_full.rst:8]: two spaces before Consider

File: docs/clients/coil_temperature_estimator_full.rst [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+convection coefficient if a relatively large or small propeller is used or if extreme performance, and thus
```
Comment (vertiq-jordan) [inline docs/clients/coil_temperature_estimator_full.rst:9]: Consider changing the convection coefficient if using a relatively large or small propeller, or if extreme performance and accuracy are required.

File: docs/clients/coil_temperature_estimator_full.rst [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+| 8      | otw                          | get, set, save | float     | degC         | Over temperature warning, Derating of the motor begins at this temperature                                              |
++--------+------------------------------+----------------+-----------+--------------+-------------------------------------------------------------------------------------------------------------------------+
+| 9      | otlo                         | get, set, save | float     | degC         | Over temperature lock out, Derating of the motor end at this temperature, where the motor is fully disabled             |
++--------+------------------------------+----------------+-----------+--------------+-------------------------------------------------------------------------------------------------------------------------+
+| 10     | derate                       | get            | float     |              | Amount of derating applied to motor [0 1] where 1 is normal operation                                                   |
```
Comment (vertiq-jordan) [inline docs/clients/coil_temperature_estimator_full.rst:139]: [0, 1]

File: docs/clients/iquart_flight_controller_interface.rst [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+        // Make a IQUART Flight Controller Interface object with obj_id 0
+        IQUartFlightControllerInterfaceClient iquartFlightControllerInterface(0);
+
+        // Use the IQUART Flight Controller Interface Client
+        iquartFlightControllerInterface.telemetry_.get(com)
+
+        // Insert code for interfacing with hardware here  
+
+        // Store telemetry data in an IFCITelemetryData object
+        IFCITelemetryData telemtry = iquart_flight_controller_interface.telemetry_.get_reply()
```
Comment (vertiq-jordan) [inline docs/clients/iquart_flight_controller_interface.rst:67]: telemtry typo

File: docs/notes/IFCITelemetryData-struct.rst [hunk 15]
```diff
@@ -0,0 +1,19 @@
+.. _ifcitelemetrydata_note:
+
+**IFCITelemetryData Struct**
+
+The data returned by the telemetry endpoint contains 16 bytes. A struct called IFCITelemtryData is required to store this data.
```
Comment (vertiq-jordan) [inline docs/notes/IFCITelemetryData-struct.rst:5]: Telemtry typo

File: docs/clients/iquart_flight_controller_interface.rst [hunk 16]
```diff
@@ -0,0 +1,132 @@
+IQUART Flight Controller Interface
+----------------------------------
+This client is used to simplify communication between the flight controller and multiple modules that are connected to an IQUART bus.
```
Comment (vertiq-jordan) [inline docs/clients/iquart_flight_controller_interface.rst:3]: used to simplify communication between flight controllers and multiple modules connected to the IQUART bus.

File: docs/clients/iquart_flight_controller_interface.rst [hunk 17]
```diff
@@ -0,0 +1,132 @@
+IQUART Flight Controller Interface
```
Comment (vertiq-jordan) [inline docs/clients/iquart_flight_controller_interface.rst:1]: maybe add (IFCI) to the end of this

File: docs/clients/pulsing_rectangular_input_parser.rst [hunk 18]
```diff
@@ -0,0 +1,112 @@
+Pulsing Rectangular Input Parser
+--------------------------------
+This client is used with pulsing modules to use the x and y coordinate commands from the flight controller and converts them into a format that can be used by the pulsing module.
```
Comment (vertiq-jordan) [inline docs/clients/pulsing_rectangular_input_parser.rst:3]: This pulsing module specific client is used to convert flight controller x/y coordinate commands into a format understood by pulsing modules.

File: docs/clients/voltage_superposition.rst [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+    |variable_name| = iq.|module_name|(com, 0)
+    
+    zero_angle= |variable_name|.get("voltage_superposition", "zero_angle") 
+    print(f"Zero angle: {zero_angle}")
+
+Message Table
+~~~~~~~~~~~~~
+
+Type ID 74 | Voltage Superposition
+
```
Comment (vertiq-jordan) [inline docs/clients/voltage_superposition.rst:105]: same thing with periods at the end of notes

File: docs/clients/voltage_superposition.rst [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
++--------+-------------------------------+----------------+-----------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 2      | phase                         | get, set       | float     | 1/s (rad)     | This value is added to the zero angle to set where the pulse occurs.                                                                                                                    |
++--------+-------------------------------+----------------+-----------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| 3      | amplitude                     | get, set       | float     | V             | This value represents the strength of the pulse. Setting the amplitude too high at lower speeds can cause the pulsing to overcome the inertia of the motor spinning, causing it to stop |
```
Comment (vertiq-jordan) [inline docs/clients/voltage_superposition.rst:115]: I think we're replacing motor verbiage with module? @iq-fred?

Comment (iq-fred) [reply docs/clients/voltage_superposition.rst:115]: In the tutorials I stuck to module when describing the whole package. In this case I don't think its a big deal, it is the motor part of the module that is really spinning here. Not sure its worth changing up everything in this tables for it.

File: docs/clients/brushless_drive.rst
[the remark this answers is not in the document]
Comment (iq-matt) [reply docs/clients/brushless_drive.rst:1]: I vote always use periods.  When a description has 2 sentences it needs them, so using them everywhere makes those 2+ descriptions not look dumb.

File: docs/clients/brushless_drive.rst (hunk 7 above)
Comment (iq-matt) [reply docs/clients/brushless_drive.rst:139]: Either ok with me.  Slight preference to removing.

File: docs/clients/coil_temperature_estimator_full.rst (hunk 10 above)
Comment (iq-matt) [reply docs/clients/coil_temperature_estimator_full.rst:137]: And Derating should be derating.  No caps needed here.

File: docs/clients/coil_temperature_estimator_full.rst [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+| 6      | c_coil                       | get, set, save | float     |              | Thermal heat capacitance/mass of the coils                                                                              |
++--------+------------------------------+----------------+-----------+--------------+-------------------------------------------------------------------------------------------------------------------------+
+| 7      | h_coil_amb_forced_conv_coeff | get, set, save | float     |              | Forced convection coefficient calculation coefficient, h_forced_conv = h_conv_coeff * sqrt(speed)                       |
++--------+------------------------------+----------------+-----------+--------------+-------------------------------------------------------------------------------------------------------------------------+
+| 8      | otw                          | get, set, save | float     | degC         | Over temperature warning, Derating of the motor begins at this temperature                                              |
```
Comment (iq-matt) [inline docs/clients/coil_temperature_estimator_full.rst:135]: derating

**Review by iq-matt — CHANGES_REQUESTED** [review]
Comments inline.  Agree with Jordan about periods.  Also check random caps.  Content looks good.

**Review by vertiq-jordan — APPROVED** [review]

**Comment by iq-fred** [comment]
@iq-matt Ben updated to address comments, take another look when you can

File: docs/_static/Vertiq_8108_white.jpg
Comment (iq-matt) [inline docs/_static/Vertiq_8108_white.jpg:1]: Yes, this picture needs to be changed.  It also needs to be cropped.  The motor is very small in the picture.

File: docs/clients/coil_temperature_estimator.rst [hunk 22]
```diff
@@ -83,9 +83,9 @@ A minimal working example for the CoilTemperatureEstimatorClient is:
     % Make a communication interface object
     com = MessageInterface(’COM18’,115200);
     % Make a CoilTemperatureEstimatorClient object with obj_id 0
-    tes = CoilTemperatureEstimatorClient(’com’,com);
+    coilTemperatureEstimator = CoilTemperatureEstimatorClient(’com’,com);
```
Comment (iq-matt) [inline docs/clients/coil_temperature_estimator.rst:86]: I agree tes is a pretty bad variable name.  
com is an object variable.
coilTemperatureEstimator is an object variable.
coil_temp is a normal variable.

The Matlab style guide says "Write variable names in mixed case starting with lower case."
The Matlab style guide says "Structure names should begin with a capital letter."
It does not say what to do with class objects.  Are they variables (they have a type of Class x)?  Are they structures (they can be . indexed)?

They have different styles.

File: docs/clients/iquart_flight_controller_interface.rst [hunk 23]
```diff
@@ -88,11 +88,12 @@ A minimal working example for the IQUartFlightControllerInterfaceClient is:
 .. code-block:: Matlab
 
     % Make a communication interface object
-    com = MessageInterface(’COM18’,115200);
-    % Make a IQUartFlightControllerInterfaceClient object with obj_id 0
-    iquartFlightControllerInterface = IQUartFlightControllerInterfaceClient(’com’,com);
-    % Use the IQUartFlightControllerInterfaceClient object
-    pulsingVoltageMode = iquartFlightControllerInterface.get(’telemetry’);
+    com = MessageInterface('COM18', 115200);
```
Comment (iq-matt) [inline docs/clients/iquart_flight_controller_interface.rst:91]: Variable/structure name convention

File: docs/clients/power_safety.rst [hunk 24]
```diff
@@ -78,13 +78,12 @@ A minimal working example for the PowerSafetyClient is:
...
 
     % Make a communication interface object
-    com = MessageInterface(’COM18’,115200);
-    
-    % Make a PowerMontitorClient object with obj_id 0
-    pwr = PowerMontitorClient(’com’,com);
-    
-    % Use the PowerMontitorClient object
-    volts = pwr.get(’volts’);
+    com = MessageInterface('COM18', 115200);
```
Comment (iq-matt) [inline docs/clients/power_safety.rst:81]: struct/variable name

File: docs/clients/propeller_motor_control.rst [hunk 25]
```diff
@@ -116,40 +116,46 @@ Message Table
...
+| 0      | ctrl_mode           | get            | int8      | enum       | -1 = no change, 0 = brake, 1 = coast, 2 = pwm, 3 = volts, 4 = velocity, 5 = thrust          |
++--------+---------------------+----------------+-----------+------------+---------------------------------------------------------------------------------------------+
+| 1      | ctrl_brake          | set            |           |            | Shorts motor leads, slows motor down dissipating energy in motor                            |
++--------+---------------------+----------------+-----------+------------+---------------------------------------------------------------------------------------------+
+| 2      | ctrl_coast          | set            |           |            | Disables all drive circuitry                                                                |
++--------+---------------------+----------------+-----------+------------+---------------------------------------------------------------------------------------------+
+| 3      | ctrl_pwm            | get, set       | float     | PWM        | [-1, 1] fraction of input voltage                                                           |
```
Comment (iq-matt) [inline docs/clients/propeller_motor_control.rst:128]: PWM or PU (-1 to 1)?

File: docs/clients/uavcan_node.rst [hunk 26]
```diff
@@ -0,0 +1,132 @@
...
+To use the UAVCAN Node client in Matlab, all Vertiq communication code must be included in your
+path. This allows the creation of a UavcanNodeClient object. See the Message Table below for available messages.
+All message strings use the Short Names. All messages use the standard Get/Set/Save functions.
+
+A minimal working example for the UavcanNodeClient is:
+
+.. code-block:: Matlab
+
+    % Make a communication interface object
+    com = MessageInterface(’COM18’,115200);
```
Comment (iq-matt) [inline docs/clients/uavcan_node.rst:75]: variable or struct name

**Review by iq-matt — APPROVED** [review]

**Pull request merged**